terraform/variables.tf

```
@@ -39,4 +39,10 @@ variable "units" {
   description = "Number of units"
   type        = number
   default     = 1
-}
\ No newline at end of file
+}
+
+variable "grafana_data_storage_size" {
+  description = "Grafana agent charm: set the size of the data volume. The default is 1 GB."
+  type        = string
+  default     = 1GB
+}
```

According to the spec:

<b style="font-weight:normal;" id="docs-internal-guid-b0ae7e34-7fff-8fd9-fca8-a6665610170d"><div dir="ltr" style="margin-left:0pt;" align="center">
Name | Type | Default Value | Description
-- | -- | -- | --
storage | map(string) | {} | Map of storage used by the application

</div></b>


```suggestion
variable "storage" {
  description = "Map of storage used by the application."
  type        = map(string)
  default     = {}
}
```

---

What is interesting to investigate here is what the TF provider tell Juju to provision when the default is used: `{}`. If we are opintionated on this by defining a static string like:

```
  default     = {
    data = "kubernetes,1,1024M"
  }
```

---

I guess we don't need to provide a default value in the terraform files, juju takes care of it. If required i can add a line in the description mentioning that the default value is 1 GB if not provided. Let me know!